popde/density_estimate.py

```diff
@@ -1,3 +1,92 @@
+import numpy as np
+
+class SimpleKernelDensityEstimation:
+    """
+    Fit and evaluate general Kernel Density Estimation (KDE) 
+
+    Methods:
+    --------
+    fit(data, bandwidth=None, kernel=None):
+        Fit the KDE to new data.
+
+    evaluate(points):
+        Evaluate the KDE at given data points.
+
+    Examples:
+    ---------
+    # Create a KDE instance and fit it to data
+    data = np.random.randn((100, 100))
+    kde = SimpleGaussianKernelDensityEstimation(data)
+
+    # Evaluate the KDE at new data points
+    new_data = np.array(np.linspace(-3, 3, 100).tolist(), np.linspace(-3, 3, 100).tolist())
+    density_values = kde.evaluate(new_data)
+    """
+
+    def __init__(self, data, bandwidth=1.0, kernel='gaussian', dim_names=None):
+        """
+        Initialize the KernelDensityEstimation object.
+
+        Parameters:
+        -----------
+        data : array-like
+            The input data for which KDE will be estimated.
+
+        bandwidth : float, optional (default=1.0)
+            Bandwidth parameter controlling the width of the kernels.
+            Larger bandwidths lead to smoother density estimates.
+
+        kernel : choice of kernel, default='gaussian'
+
+        dim_names : list of the dimension names (m1,z, chi_eff) 
+             If dim_names is not Nonecheck that the dimensionality
+             of the data array matches with len(dim_names)
```

Comment:
```suggestion
        dim_names : sequence of dimension names, e.g. ('m1', 'z', 'chi_eff') 
```
You should just document the kwarg here, put a comment at the dimensionality check method instead

Reply:
(also, the names have to be strings!